### PR TITLE
fix: agenda

### PR DESCRIPTION
**Pull request by Prerna290**
# PR Template

## Type

- (X) Fix
- ( ) Refactor
- ( ) Style
- ( ) Docs
- ( ) Feat
- ( ) Hotfix

### CODE REVIEW

File: apps/commudle-admin/src/app/feature-modules/events/components/event-embedded-video-stream/event-embedded-video-stream.component.scss [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .embedded-video-stream {
+  @apply com-bg-white;
   display: flex;
   flex-direction: column;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/events/components/event-embedded-video-stream/event-embedded-video-stream.component.scss]: let's convert this into tailwind css

File: libs/shared/services/src/lib/api-routes.constant.ts [hunk 2]
```diff
@@ -404,6 +404,8 @@ export const API_ROUTES = {
       LIVE_EVENT_SESSIONS: 'api/v2/track_slots/public_live_event_sessions',
```
Comment (ArshdeepGrover) [inline libs/shared/services/src/lib/api-routes.constant.ts]: Let's make a proper order
 TRACK_SLOTS_BY_DATE: 'api/v2/track_slots/track_slots_by_date',
    TRACK_SLOTS_BY_DATE_AND_LOCATION: 'api/v2/track_slots/track_slot_by_date_and_location',
    
    this both will be before public object

File: apps/shared-models/locations.model.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ import { ILocation } from './location.model';
 
 export interface ILocations {
   locations: ILocation[];
+  date?: Date;
```
Comment (ArshdeepGrover) [inline apps/shared-models/locations.model.ts]: why dowe use "?" in date ?

File: apps/shared-models/event-location.model.ts [hunk 4]
```diff
@@ -11,9 +11,13 @@ export interface IEventLocation {
   event_type: string;
 }
 
+export interface IEventDatesLocation {
+  date: Date;
+  event_locations: IEventLocation[];
+}
```
Comment (ArshdeepGrover) [inline apps/shared-models/event-location.model.ts:18]: let's reconfirm with event_location and location model

File: apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-locations.component.html [hunk 5]
```diff
@@ -172,9 +187,9 @@
           <nb-option [value]="EEmbeddedVideoStreamSources.OTHER"> Other (Embedded Iframe) </nb-option>
           <nb-option [value]="EEmbeddedVideoStreamSources.ZOOM" disabled> Zoom </nb-option>
         </nb-select>
-      </div>
+      </div> -->
 
-      <div>
+      <!-- <div>
         <label>
           <span
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-locations.component.html]: let's make a single comment for this

File: apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/track-slot-form/track-slot-form.component.html [hunk 6]
```diff
@@ -21,19 +21,19 @@
...
             >
-              <nb-option *ngIf="eventLocations.length === 0 || eventLocations[0].event_location_tracks.length === 0">
+              <!-- <nb-option *ngIf="eventLocations.length === 0 || eventLocations[0].event_location_tracks.length === 0">
                 <a
                   [routerLink]="['./admin/communities', community.slug, 'event-dashboard', event.slug, 'agenda']"
                   (click)="close()"
                   class="com-text-blac com-no-underline"
                   >+ Add Location /Tracks</a
                 >
+              </nb-option> -->
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/track-slot-form/track-slot-form.component.html]: let's test this for all edge cases

File: apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts [hunk 7]
```diff
@@ -464,4 +486,44 @@ export class EventLocationTracksComponent implements OnInit {
...
+    });
+  }
+  // embeddedVideoStream
+  updateEmbededContent(embeddedFormData, eventLocationTrack, elti) {
+    // this.embeddedFormData = embeddedFormData;
+    this.eventLocationTracks[elti].embedded_video_stream = embeddedFormData;
+    this.dialogRef.close();
+    this.editTrack(eventLocationTrack.id, embeddedFormData, eventLocationTrack);
+  }
 }
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts]: remove comment if not required!

File: apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts [hunk 8]
```diff
@@ -40,22 +41,17 @@ import * as moment from 'moment';
...
 })
-export class EventLocationTracksComponent implements OnInit {
+export class EventLocationTracksComponent implements OnInit, OnChanges {
   @Input() eventLocations: IEventLocation[] = [];
-  @Input() eventLocationTracks: IEventLocationTrack[] = [];
+  @Input() sessionDate: Date;
   @Input() event: IEvent;
   @Input() community: ICommunity;
-  @Input() eventLocationId;
+  @Input() eventLocation;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts]: let's give a data type IEventLocation

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

**Review by ArshdeepGrover — APPROVED** [review]
